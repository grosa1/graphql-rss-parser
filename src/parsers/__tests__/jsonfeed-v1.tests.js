/* eslint-env jest */
const parse = require('../jsonfeed-v1')
const mockAxios = require('../../__tests__/mockAxios')

test('should parse from rolflekang.com/feed.json', async () => {
  const fixture = await mockAxios({ url: 'https://rolflekang.com/feed.json' })

  expect(await parse(fixture.data)).toMatchSnapshot()
})
