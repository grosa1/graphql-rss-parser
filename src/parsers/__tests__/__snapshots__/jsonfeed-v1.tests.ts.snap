// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse from rolflekang.com/feed.json 1`] = `
Object {
  "authors": Array [
    Object {
      "name": "Rolf Erik Lekang",
    },
  ],
  "description": undefined,
  "feed_url": "https://rolflekang.com/feed.json",
  "home_page_url": "https://rolflekang.com",
  "items": Array [
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2020-11-08T00:00:00.000Z",
      "date_published": "2020-11-08T00:00:00.000Z",
      "id": "https://rolflekang.com/using-certbot-with-ansible",
      "tags": Array [
        "letsencrypt",
        "acme",
        "certbot",
        "ansible",
      ],
      "title": "Using certbot with Ansible",
      "url": "https://rolflekang.com/using-certbot-with-ansible",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2020-09-12T00:00:00.000Z",
      "date_published": "2020-09-12T00:00:00.000Z",
      "id": "https://rolflekang.com/ansible-handlers-in-loops",
      "tags": Array [
        "ansible",
        "devops",
        "ops",
        "infrastructure-as-code",
      ],
      "title": "Using Ansible handlers in loops",
      "url": "https://rolflekang.com/ansible-handlers-in-loops",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2020-05-26T00:00:00.000Z",
      "date_published": "2020-05-26T00:00:00.000Z",
      "id": "https://rolflekang.com/serving-plain-text-with-nextjs",
      "tags": Array [
        "nextjs",
        "javascript",
        "curl",
        "nodejs",
      ],
      "title": "Serving text/plain for curl with Next",
      "url": "https://rolflekang.com/serving-plain-text-with-nextjs",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2020-05-03T00:00:00.000Z",
      "date_published": "2020-05-03T00:00:00.000Z",
      "id": "https://rolflekang.com/wireless-uplinks-with-unifi",
      "tags": Array [
        "networking",
        "unifi",
        "ubiquiti",
        "homelab",
      ],
      "title": "Wireless uplinks with Unifi",
      "url": "https://rolflekang.com/wireless-uplinks-with-unifi",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2019-11-14T00:00:00.000Z",
      "date_published": "2019-11-14T00:00:00.000Z",
      "id": "https://rolflekang.com/using-git-commits-instead-of-stash",
      "tags": Array [
        "git",
      ],
      "title": "Using git commits instead of git stash",
      "url": "https://rolflekang.com/using-git-commits-instead-of-stash",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2019-02-26T00:00:00.000Z",
      "date_published": "2019-02-26T00:00:00.000Z",
      "id": "https://rolflekang.com/twitter-cards-for-gatsby-posts",
      "tags": Array [
        "gatsby",
        "canvas",
        "javascript",
      ],
      "title": "Twitter cards for Gatsby posts",
      "url": "https://rolflekang.com/twitter-cards-for-gatsby-posts",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2019-02-20T00:00:00.000Z",
      "date_published": "2019-02-20T00:00:00.000Z",
      "id": "https://rolflekang.com/related-articles-with-gatsby",
      "tags": Array [
        "gatsby",
        "javascript",
      ],
      "title": "Related articles with Gatsby",
      "url": "https://rolflekang.com/related-articles-with-gatsby",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2019-02-12T00:00:00.000Z",
      "date_published": "2019-02-12T00:00:00.000Z",
      "id": "https://rolflekang.com/creating-a-cli-with-reason-native",
      "tags": Array [
        "reasonml",
        "native",
      ],
      "title": "Creating a CLI with Reason native",
      "url": "https://rolflekang.com/creating-a-cli-with-reason-native",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2017-05-28T00:00:00.000Z",
      "date_published": "2017-05-28T00:00:00.000Z",
      "id": "https://rolflekang.com/testing-simple-graphql-services",
      "tags": Array [
        "graphql",
        "testing",
        "javascript",
      ],
      "title": "Testing simple GraphQL services",
      "url": "https://rolflekang.com/testing-simple-graphql-services",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2017-01-02T00:00:00.000Z",
      "date_published": "2017-01-02T00:00:00.000Z",
      "id": "https://rolflekang.com/lets-run-some-races",
      "tags": Array [
        "running",
      ],
      "title": "Let's run some races this year",
      "url": "https://rolflekang.com/lets-run-some-races",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2017-01-01T00:00:00.000Z",
      "date_published": "2017-01-01T00:00:00.000Z",
      "id": "https://rolflekang.com/software-development-on-an-ipad",
      "tags": Array [
        "ipad",
        "development-environment",
      ],
      "title": "Software development on an iPad",
      "url": "https://rolflekang.com/software-development-on-an-ipad",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2016-06-10T00:00:00.000Z",
      "date_published": "2016-06-10T00:00:00.000Z",
      "id": "https://rolflekang.com/filtering-lint-errors",
      "tags": Array [
        "linter",
        "linting",
        "development",
      ],
      "title": "Filtering lint errors",
      "url": "https://rolflekang.com/filtering-lint-errors",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2015-07-26T00:00:00.000Z",
      "date_published": "2015-07-26T00:00:00.000Z",
      "id": "https://rolflekang.com/interrail-summer",
      "tags": Array [
        "travel",
        "interrail",
      ],
      "title": "Interrail summer",
      "url": "https://rolflekang.com/interrail-summer",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2015-05-23T00:00:00.000Z",
      "date_published": "2015-05-23T00:00:00.000Z",
      "id": "https://rolflekang.com/writing-latex-in-atom",
      "tags": Array [
        "atom",
        "writing",
        "procrastination",
      ],
      "title": "Writing (latex) in Atom",
      "url": "https://rolflekang.com/writing-latex-in-atom",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2014-12-30T00:00:00.000Z",
      "date_published": "2014-12-30T00:00:00.000Z",
      "id": "https://rolflekang.com/rmoq",
      "tags": Array [
        "python",
        "testing",
        "mocking",
      ],
      "title": "rmoq - a request mock cache",
      "url": "https://rolflekang.com/rmoq",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2014-10-03T00:00:00.000Z",
      "date_published": "2014-10-03T00:00:00.000Z",
      "id": "https://rolflekang.com/django-nopassword-one-point-o",
      "tags": Array [
        "django",
        "nopassword",
        "django-nopassword",
        "authentication",
      ],
      "title": "django-nopassword reached 1.0",
      "url": "https://rolflekang.com/django-nopassword-one-point-o",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2014-05-07T00:00:00.000Z",
      "date_published": "2014-05-07T00:00:00.000Z",
      "id": "https://rolflekang.com/moving-to-pelican",
      "tags": Array [
        "this",
        "static-site",
        "pelican",
        "jekyll",
      ],
      "title": "Moving the blog to pelican",
      "url": "https://rolflekang.com/moving-to-pelican",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2014-02-02T00:00:00.000Z",
      "date_published": "2014-02-02T00:00:00.000Z",
      "id": "https://rolflekang.com/building-tumblr-theme-with-grunt",
      "tags": Array [
        "gruntjs",
        "cookiecutter",
        "tumblr",
      ],
      "title": "Building a Tumblr theme with GruntJS",
      "url": "https://rolflekang.com/building-tumblr-theme-with-grunt",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2013-10-24T00:00:00.000Z",
      "date_published": "2013-10-24T00:00:00.000Z",
      "id": "https://rolflekang.com/backup-routines-for-redisdb",
      "tags": Array [
        "db",
        "backup",
        "redis",
      ],
      "title": "Backup routine for Redis",
      "url": "https://rolflekang.com/backup-routines-for-redisdb",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2013-05-22T00:00:00.000Z",
      "date_published": "2013-05-22T00:00:00.000Z",
      "id": "https://rolflekang.com/django-app-custom-user",
      "tags": Array [
        "django",
        "custom user",
      ],
      "title": "Making your Django app ready for custom users",
      "url": "https://rolflekang.com/django-app-custom-user",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2013-03-03T00:00:00.000Z",
      "date_published": "2013-03-03T00:00:00.000Z",
      "id": "https://rolflekang.com/stats-are-fun",
      "tags": Array [
        "pypstats",
        "pypi",
        "stats",
        "geek-tool",
      ],
      "title": "Stats are fun",
      "url": "https://rolflekang.com/stats-are-fun",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2013-02-26T00:00:00.000Z",
      "date_published": "2013-02-26T00:00:00.000Z",
      "id": "https://rolflekang.com/alfred-scripts-for-jekyll",
      "tags": Array [
        "alfredapp",
        "jekyll",
        "shell script",
      ],
      "title": "Alfred scripts for Jekyll",
      "url": "https://rolflekang.com/alfred-scripts-for-jekyll",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2013-01-27T00:00:00.000Z",
      "date_published": "2013-01-27T00:00:00.000Z",
      "id": "https://rolflekang.com/use-array-in-postgresql",
      "tags": Array [
        "tags",
        "postgres",
        "array",
        "pgarray",
        "django",
      ],
      "title": "Use array in postgresql with django",
      "url": "https://rolflekang.com/use-array-in-postgresql",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2013-01-03T00:00:00.000Z",
      "date_published": "2013-01-03T00:00:00.000Z",
      "id": "https://rolflekang.com/apply-puppet-configuration-automatically",
      "tags": Array [
        "puppet",
        "web.py",
        "nginx",
        "uwsgi",
        "github",
        "github-hooks",
      ],
      "title": "Apply puppet automatically",
      "url": "https://rolflekang.com/apply-puppet-configuration-automatically",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2012-12-02T00:00:00.000Z",
      "date_published": "2012-12-02T00:00:00.000Z",
      "id": "https://rolflekang.com/readable-wikipedia",
      "tags": Array [
        "wikipedia",
        "css",
        "readability",
        "stylebot",
      ],
      "title": "Readable Wikipedia",
      "url": "https://rolflekang.com/readable-wikipedia",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2012-09-28T00:00:00.000Z",
      "date_published": "2012-09-28T00:00:00.000Z",
      "id": "https://rolflekang.com/django-nopassword",
      "tags": Array [
        "django",
        "no-password",
        "email-authentication",
        "authentication",
      ],
      "title": "django-nopassword",
      "url": "https://rolflekang.com/django-nopassword",
    },
    Object {
      "authors": Array [
        Object {
          "name": "Rolf Erik Lekang",
        },
      ],
      "date_modified": "2012-09-16T00:00:00.000Z",
      "date_published": "2012-09-16T00:00:00.000Z",
      "id": "https://rolflekang.com/the-github-lamp",
      "tags": Array [
        "arduino",
        "github",
        "git",
        "web.py",
      ],
      "title": "The Github lamp",
      "url": "https://rolflekang.com/the-github-lamp",
    },
  ],
  "parser": "JSON_FEED_V1",
  "title": "Writing by Rolf Erik Lekang",
}
`;
